### PR TITLE
Fixed list unpacking issue

### PR DESCRIPTION
**Pull request by MannarAmuthan**
Fix for issue: https://github.com/dictu-lang/Dictu/issues/421 

Found that we need to add separate statement parsing logic for already declared variables to list unpacking .

[X] -  Added this scenario in tests

### CODE REVIEW

**Comment by MannarAmuthan** [comment]
Found some issue, fixing it

**Comment by MannarAmuthan** [comment]
I still get the error on this example tests, even when I revert my changes

**Comment by Jason2605** [comment]
Yeah it's due to the fact that we can't use list literals. This issue is actually going to be rather tricky to solve I think

Screenshot 2022-10-20 at 17 36 26

File: tests/variables/list-unpacking.du [hunk 1]
```diff
@@ -9,6 +9,20 @@ class TestListUnpacking < UnitTest {
...
+        var o,i,g;
+
+        [o,i,g]=[111,222,333];
+
+        assert(o == 111);
+        assert(i == 222);
+        assert(g == 333);
+
+        var [d, e, f] = [1 + 1, 2 + 2, 3 + 3];
+
```
Comment (Jason2605) [inline tests/variables/list-unpacking.du:25]: We want to use the https://dictu-lang.com/docs/standard-lib/unittest/ builtin where possible over `assert` as `assert` will stop execution on first failure (as the main reason - UnitTest library has a nicer output too).

**Comment by MannarAmuthan** [comment]
This is Done

**Comment by Jason2605** [comment]
Awesome bit of work @MannarAmuthan, thanks very much for this!

**Pull request merged**